### PR TITLE
Console ANSI codes support

### PR DESCRIPTION
**Pull request by jponge**
See #318

### CODE REVIEW

**Comment by jponge** [comment]
This is a draft at this stage.

File: src/main/golo/ansicodes.golo [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+All function names are self-explanatory. Note that `ansi_likely_supported` tests if the current
+operating system is _likely_ to support ANSI codes. It merely checks that the host operating
+system is not MS Windows.
+
+See also: [Wikipedia on ANSI escape codes](https://en.wikipedia.org/wiki/ANSI_escape_code)
+----
+module gololang.AnsiCodes
+
+function ansi_likely_supported = ->
+  not System.getProperty("os.name"): contains("Windows")
```
Comment (danielpetisme) [inline src/main/golo/ansicodes.golo]: why so much hate for windows
https://github.com/adoxa/ansicon
personally I use it with cygwin, it works like a charm

Comment (jponge) [inline src/main/golo/ansicodes.golo]: This needs a driver :smile:

**Comment by Artpej** [comment]
Long time ago I've played with this : https://github.com/mabe02/lanterna. Very nice terminal abstraction, it can be useful for this one or inspiration.

**Comment by jponge** [comment]
Nice API (warning: it’s LGPL’d).

It’s probably more than what we need, as I was thinking about the very basics of ANSI codes.

**Comment by jponge** [comment]
Is the current implementation ok?

**Comment by jponge** [comment]
Also ping @k33g (forgot your gist URL 🤔)

**Comment by jponge** [comment]
Also ping @yloiseau, I know you love terminals :smile:

**Comment by jponge** [comment]
The Eclipse ip-check validation failed on my email, strange as it is the good one.

**Comment by Artpej** [comment]
@jponge  just amend your last commit with `-s` and re-push the branch should resolve the problem.

**Pull request merged**